service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -37,22 +43,34 @@
     protected boolean linkOpenedRemotely;
 
     protected final String hostName;
-    protected final String userName;
-    protected final String sasToken;
+    protected String userName;
```

Comment:
looks like we still set this only at the constructor level, could still be final

Reply:
Not every constructor sets a value for username or sasToken anymore, so it can't be final.

Reply:
got it!